projects/plugins/protect/src/js/state/actions.js

```diff
@@ -8,6 +14,29 @@ const setStatus = status => {
 	return { type: SET_STATUS, status };
 };
 
+/**
+ * Side effect action which will fetch the status from the server
+ *
+ * @returns {Promise} - Promise which resolves when the status is refreshed from an API fetch.
+ */
+const refreshStatus = () => async ( { dispatch } ) => {
+	dispatch( setStatusIsFetching( true ) );
+	return await new Promise( ( resolve, reject ) => {
+		return apiFetch( {
```

Comment:
You don't need to do all the sequence of `promises`, since you're using `async` keyword:

```es6
try {
  dispatch( setStatusIsFetching( true ) );
  const status = await apiFetch( { ... } );
  dispatch( setStatus( camelize( status ) ) );
  dispatch( setStatusIsFetching( false ) );
  return Promise.resolve( status );
} catch (error) {
  return Promise.reject( error );
}
```